types/redux-form/lib/reduxForm.d.ts

```diff
@@ -34,7 +34,7 @@ export interface Config<FormData extends DataShape, P, S> {
      * the name of your form and the key to where your form's state will be
      * mounted under the redux-form reducer
      */
-    form: string;
+    form?: string;
```

Comment:
It is required but can be defined both as configured property or by passing a property directly to the component as it states in the docs: `IMPORTANT: All of these configuration options may be passed into reduxForm() at "design time" or passed in as props to your component at runtime.`.